src/main/js/controls/Bubble/helpers/helpers.js

```diff
@@ -0,0 +1,523 @@
+    }
+
+    const renderDataPoint = (path, value, index) => {
+        path.append("g")
+            .classed(styles.point, true)
+            .attr("aria-disabled", !utils.isDefined(value.onClick))
+            .attr("transform", transformPoint(scale)(value))
+            .attr("aria-describedby", `${value.key}`)
+            .attr("aria-hidden", (value) =>
+                shouldHideDataPoints(config.shownTargets, value)
+            )
+            .on("click", function() {
+                dataPointActionHandler(value, index, d3.select(this).node());
+            })
+            .append("circle")
+            .attr("aria-describedby", value.key)
+            .attr("r", (d) => {
+                if (
+                    areWeightDefined(dataTarget) &&
+                    d.weight !== undefined &&
+                    !utils.isDefined(dataTarget.weight.maxRadius)
+                ) {
+                    return bubbleScale(d.weight);
+                }
+                if (utils.isUndefined(dataTarget.weight)) {
+                    return constants.DEFAULT_BUBBLE_RADIUS_MAX;
+                }
+                return dataTarget.weight.maxRadius;
+            })
+            .style("fill", (d) => {
+                if (
+                    isHueDefined(dataTarget.hue) &&
+                    areWeightDefined(dataTarget) &&
+                    useYAxisData(d.weight) === false
+                ) {
+                    const colorHue = generateColor(
+                        dataTarget.hue.lowerShade,
+                        dataTarget.hue.upperShade,
+                        dataTarget,
+                        useYAxisData(d.weight)
+                    );
+                    return colorHue(bubbleScale(d.weight));
+                }
+                if (
+                    isHueDefined(dataTarget.hue) &&
+                    areWeightDefined(dataTarget) === false &&
+                    useYAxisData(d.weight)
+                ) {
+                    const colorHue = generateColor(
+                        dataTarget.hue.lowerShade,
+                        dataTarget.hue.upperShade,
+                        dataTarget,
+                        useYAxisData(d.weight)
+                    );
+                    return colorHue(d.y);
+                }
+                return dataTarget.color;
+            })
+            .style("opacity", constants.DEFAULT_BUBBLE_OPACITY)
+            .attr("stroke", constants.DEFAULT_BUBBLE_STROKE);
+    };
+
+    const renderSelectionPath = (path, value, index) => {
+        path.append("g")
+            .classed(styles.dataPointSelection, true)
+            .attr("transform", transformPoint(scale)(value))
+            .attr("aria-disabled", utils.isDefined(value.onClick))
+            .attr("aria-hidden", true)
+            .attr("aria-describedby", value.key)
+            .append("circle")
+            .attr("r", (d) => {
+                if (
+                    areWeightDefined(dataTarget) &&
```

Comment:
You can do it similar to the comment in `renderDataPoint` or move the decision making parts (`if-elseif-else`) out into a function and use them in both places.

Reply:
Yes, I will make the change